daemon/algod/api/server/v2/handlers.go

```diff
@@ -651,6 +651,35 @@ func (v2 *Handlers) GetBlock(ctx echo.Context, round uint64, params model.GetBlo
 	return ctx.Blob(http.StatusOK, contentType, data)
 }
 
+// GetBlockTxids gets all top level TxIDs in a block for the given round.
+// (GET /v2/blocks/{round}/txids)
+func (v2 *Handlers) GetBlockTxids(ctx echo.Context, round uint64) error {
+	ledger := v2.Node.LedgerForAPI()
+	block, _, err := ledger.BlockCert(basics.Round(round))
+	if err != nil {
+		switch err.(type) {
+		case ledgercore.ErrNoEntry:
+			return notFound(ctx, err, errFailedLookingUpLedger, v2.Log)
+		default:
+			return internalError(ctx, err, errFailedLookingUpLedger, v2.Log)
```

Comment:
That might be from the auth middleware.

Reply:
sorry, I meant error code 400 - Bad Request - Non integer number. could we add unit tests to validate that 400 and 401 errors are returned as expected?

Reply:
`round` is a `uint64` so I think it will always be an integer number. I added a case where the round is not found in the ledger as a 404 -- is that the sort of unit test we are looking for?

Reply:
The 401 seems to be handled in our middleware, and there is a general expect test for it here: https://github.com/algorand/go-algorand/blob/master/test/e2e-go/cli/goal/expect/goalNodeConnectionTest.exp#L35 

The 400 -- I copied it from the existing BlockHash API here: https://github.com/algorand/go-algorand/pull/4580/files#diff-4dbbc679cc9edab4510c8be1d215ec85830b5a5cfba57ac1379501cbeb3911d0R550

@winder do you have an idea if this is necessary or can be removed?

Reply:
This (400) looks like it is handled independently in the server as well, and used in the Block and BlockHash endpoint, so I'm in favor of keeping it consistent with those for now. @shiqizng are we okay to resolve this thread?